src/treq/_utils.py

```diff
@@ -45,3 +48,50 @@ def default_pool(reactor, pool, persistent):
         set_global_pool(HTTPConnectionPool(reactor, persistent=True))
 
     return get_global_pool()
+
+
+def warn(*args, **kwargs):
+    """
+    Utility function to raise warnings with stack level set automatically to
+    the first position outside a given namespace.
+
+    Notice that if you pass the stacklevel keyword, it will be increased by 1
+    in order to ignore this wrapper function and that will be passed directly
+    to warnings.warn. That is: it is functionally equivalent to warnings.warn.
+
+    @param namespace: calls within this namespace (aka startswith) will not
+        be accounted for in the stacklevel variable if possible.
+        If not defined, the default is the top level package of this code.
+    @param default_stacklevel: in those cases in which there is no code
+        outside of namespace in the stack, we will default to this value.
+        This utility function already takes into account the fact that it is a
+        wrapper around warnings.warn and secretly passes default_stacklevel+1
+        when necessary.
+        Defaults to 1, which would match the place where this was called from.
+    """
+    # Default to top-level package
+    namespace = kwargs.pop('namespace', __name__.partition('.')[0])
+    default_stacklevel = kwargs.pop('default_stacklevel', 1)
```

Comment:
Good golly this review is scattered...  I should just stop for now.  You are claiming the `namespace` (should this be `top_level_package`?  even your comment says that) and `default_stacklevel` parameters.  Perhaps there should be tests verifying that they are not valid for `warnings.warn()`.  Just in case they get added someday.  Or maybe that's excessive.  I'm not sure offhand how to write them without just duplicating the names which hazards the test and this function getting out of sync.